### PR TITLE
concurrently load replicas on init, refer to #491

### PR DESCRIPTION
**Pull request by qinzuoyan**

### CODE REVIEW

File: src/dist/replication/lib/replica_stub.cpp [hunk 1]
```diff
@@ -170,6 +172,10 @@ void replica_stub::initialize(const replication_options& opts, bool clear/* = fa
         dir_list.insert(dir_list.end(), tmp_list.begin(), tmp_list.end());
     }
 
+    replicas rps;
+    utils::ex_lock rps_lock;
```
Comment (imzhenyu) [inline src/dist/replication/lib/replica_stub.cpp:176]: We should use service::zlock instead to enable tooling unless it is really performance critical for the lock overhead.

**Comment by imzhenyu** [comment]
@qinzuoyan I will merge this one for now. I'm wondering whether you have future changes to the core part, as that will be not easy to merge any more due to the directory name change done in #494 ?

**Comment by qinzuoyan** [comment]
@imzhenyu, yes, there are still some bug found by kill_test.

**Comment by imzhenyu** [comment]
Then you may have to merge them manually instead of cherry-pick later as the organization of the directories are changed in #494 and after. I'll merge #494 later today in case you guys have some important fixes done today.

**Pull request merged**